yb-voyager/cmd/importData.go

```diff
@@ -1206,6 +1223,33 @@ func extractCopyStmtForTable(table string, fileToSearchIn string) {
 	}
 }
 
+// get the parent table if its a partitioned table in yugabytedb
+// also covers the case when the partitions are further subpartitioned to multiple levels
+func getParentTable(table string, conn *pgx.Conn) string {
+	var parentTable string
+	query := fmt.Sprintf(`SELECT inhparent::pg_catalog.regclass
+	FROM pg_catalog.pg_class c JOIN pg_catalog.pg_inherits i ON c.oid = inhrelid
+	WHERE c.oid = '%s'::regclass::oid`, table)
+
+	var currentParentTable string
+	err := conn.QueryRow(context.Background(), query).Scan(&currentParentTable)
+	if err != pgx.ErrNoRows && err != nil {
+		utils.ErrExit("Error in querying parent tablename for table=%s: %v", table, err)
+	}
+
+	if len(currentParentTable) == 0 {
+		return ""
+	} else {
+		parentTable = getParentTable(currentParentTable, conn)
+		if len(parentTable) == 0 {
+			parentTable = currentParentTable
+		}
+	}
+
+	log.Infof("parentTable=%s for table=%s", parentTable, table)
```

Comment:
We can probably put this log statement outside the recursion of the function, it could otherwise lead to a lot of unnecessary cluttering in the log file; log the "oldest" parent of the table we wish to query

Reply:
done